File: part2/src/DistributedPuzzle_Actors/src/main/scala/GUI/Tile.scala
package GUI

import java.awt.Image

case class Tile(var image: Image, var originalPosition: Int, var currentPosition: Int) extends Comparable[Tile] with Serializable:

    def getImage: Image = image

    def isInRightPlace: Boolean = currentPosition == originalPosition

    def getCurrentPosition: Int = currentPosition

    def setCurrentPosition(newPosition: Int): Unit = currentPosition = newPosition

    override def compareTo(other: Tile): Int = other.getCurrentPosition match
        case x if x > this.currentPosition => -1
        case x if x == this.currentPosition => 0
        case _ => 1

File: part2/JavaRMI/src/main/java/org/example/puzzle/Tile.java
package org.example.puzzle;

import java.awt.Image;

public class Tile implements Comparable<Tile> {
	private final Image image;
	private final int originalPosition;
	private int currentPosition;

    public Tile(final Image image, final int originalPosition, final int currentPosition) {
        this.image = image;
        this.originalPosition = originalPosition;
        this.currentPosition = currentPosition;
    }

    public int getOriginalPosition() {
        return originalPosition;
    }

    public Image getImage() {
    	return image;
    }
    
    public boolean isInRightPlace() {
    	return currentPosition == originalPosition;
    }
    
    public int getCurrentPosition() {
    	return currentPosition;
    }
    
    public void setCurrentPosition(final int newPosition) {
    	currentPosition = newPosition;
    }

	@Override
	public int compareTo(Tile other) {
		return this.currentPosition < other.currentPosition ? -1
				: (this.currentPosition == other.currentPosition ? 0 : 1);
	}
}
